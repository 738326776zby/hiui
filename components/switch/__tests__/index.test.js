import React from 'react'
import { mount } from 'enzyme'
import renderer from 'react-test-renderer'
import Switch from '..'

describe('Switch', () => {
  beforeAll(() => {
    jest.useFakeTimers()
  })

  afterAll(() => {
    jest.useRealTimers()
  })
  it('basic use', () => {
    const wrapper = renderer.create(
      <div>
        <Switch />
        <Switch content={['ON', 'OFF']} />
        <Switch checked disabled content={['开', '关']} />
      </div>
    )
    expect(wrapper).toMatchSnapshot()
  })
  it('onchange', () => {
    const onChange = jest.fn()
    const wrapper = mount(<Switch onChange={onChange} />)
    wrapper.find('.hi-switch').simulate('click')
    expect(onChange).toBeCalled()
    wrapper.setProps({ checked: true })
    expect(onChange).toHaveBeenCalledTimes(2)
    wrapper.setProps({ uselessProps: true })
    expect(onChange).toHaveBeenCalledTimes(2)
  })

  it('should stop when click return false', () => {
    const onChange = jest.fn()
    const onClick = jest.fn(state => false)
    const wrapper = mount(<Switch onChange={onChange} onClick={onClick} />)
    wrapper.find('.hi-switch').simulate('click')
    expect(onChange).not.toBeCalled()
  })

  it('should be disabled', () => {
    const onChange = jest.fn()
    const wrapper = mount(<Switch onChange={onChange} disabled />)
    wrapper.find('.hi-switch').simulate('click')
    expect(onChange).not.toBeCalled()
  })
})
