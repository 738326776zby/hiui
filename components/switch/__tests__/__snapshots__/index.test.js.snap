// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Switch basic use 1`] = `
<div>
  <span
    className="theme__hiui-blue hi-switch hi-switch--closed"
    onClick={[Function]}
  >
    <span
      className="hi-switch__text"
    />
  </span>
  <span
    className="theme__hiui-blue hi-switch hi-switch--closed"
    onClick={[Function]}
  >
    <span
      className="hi-switch__text"
    >
      OFF
    </span>
  </span>
  <span
    className="theme__hiui-blue hi-switch hi-switch--disabled"
    onClick={[Function]}
  >
    <span
      className="hi-switch__text"
    >
      å¼€
    </span>
  </span>
</div>
`;
